File: 프로그래머스/lv1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {

        int[] oneArr = {1, 2, 3, 4, 5};
        int[] twoArr = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] threeArr = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int[] score = {0, 0, 0};
        
        for (int i = 0; i < answers.length; i++) {
            if (answers[i] == oneArr[i % 5]) score[0]++;
            if (answers[i] == twoArr[i % 8]) score[1]++;
            if (answers[i] == threeArr[i % 10]) score[2]++;
        }
        
        int max = Math.max(score[0], Math.max(score[1], score[2]));
        List<Integer> list = new ArrayList<>();
        for (int i = 0; i < score.length; i++) {
            if (max == score[i]) list.add(i+1);
        }
        
        int[] answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }
        
        
        return answer;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):

    spj1 = [1,2,3,4,5]
    spj2 = [2, 1, 2, 3, 2, 4, 2, 5]
    spj3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    rank = [0, 0, 0, 0]
    
    for i in range(len(answers)):
        if spj1[i % len(spj1)] == answers[i]:
            rank[1] += 1
        if spj2[i % len(spj2)] == answers[i]:
            rank[2] += 1
        if spj3[i % len(spj3)] == answers[i]:
            rank[3] += 1
    
    answer = [i for i in range(1, len(rank)) if rank[i] == max(rank)]
    
    return answer